e2e-tests/cypress/support/helpers/SmartKbnApiClient.ts

```diff
@@ -0,0 +1,27 @@
+import { KbnApiClient } from './KbnApiClient';
+
+export class SmartKbnApiClient extends KbnApiClient {
+
+  public static instance: SmartKbnApiClient = new SmartKbnApiClient();
```

Comment:
Why do we prefer this apporach and use like `SmartKbnApiClient.instance.deleteSavedObjects()` over `SmartKbnApiClient.deleteSavedObjects()`?

Reply:
I meant here, that this page object is just an abstraction that wraps multiple cypress methods and gives them a meaningful name and generally I don't see a profit to create a new instance of the class (it's a huge chance that I don't get it :) ), but if we want to have it, we can do something like

```
class SmartKbnApiClient extends KbnApiClient {
...
}
...


export const smartKbnApiClient = new SmartKbnApiClient()

```
and use it like

```
smartKbnApiClient.deleteSavedObjects()
```

Reply:
Because it's a class. You have to have an instance. It's class because I wanted KbnApiAdvancedClient to extend KbnApiClient. Thanks to this, you can call methods from KbnApiClient using KbnApiAdvancedClient instance.

With a static module I don't know how to achieve the above.

Reply:
Ohh, you are right 🤦 , there is no way to do it in case of a static approach, still we can do it 

```
class SmartKbnApiClient extends KbnApiClient {
...
}
...


export const smartKbnApiClient = new SmartKbnApiClient()

and use it like

smartKbnApiClient.deleteSavedObjects()
```

and don't bother the developer with additional `instance`, but it's up to you, it's a non-blocking comment, you can resolve it

Reply:
I forgot that you can export const from whatever you want :D 
this is better, I've changed it.